File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/strategy/LinearRegressionExtrapolationStrategy.java
package org.kevoree.modeling.api.strategy;

import org.kevoree.modeling.api.KObject;
import org.kevoree.modeling.api.meta.MetaAttribute;

/**
 * Created by duke on 10/28/14.
 */
public class LinearRegressionExtrapolationStrategy implements ExtrapolationStrategy {
    @Override
    public Long[] timedDependencies(KObject current) {
        //TODO ASSAD HERE, thinks about the case that the object
        return new Long[0];
    }

    @Override
    public Object extrapolate(KObject current, MetaAttribute attribute, KObject[] dependencies) {

        return null;
    }

    @Override
    public void mutate(KObject current, MetaAttribute attribute, Object payload, KObject[] dependencies) {
        
    }
}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/strategy/LinearRegressionExtrapolationStrategy.ts
///<reference path="../KObject.ts"/>
///<reference path="../meta/MetaAttribute.ts"/>

class LinearRegressionExtrapolationStrategy implements ExtrapolationStrategy {

  public timedDependencies(current: KObject<any,any>): number[] {
    return new Array();
  }

  public extrapolate(current: KObject<any,any>, attribute: MetaAttribute, dependencies: KObject<any,any>[]): any {
    return null;
  }

  public mutate(current: KObject<any,any>, attribute: MetaAttribute, payload: any, dependencies: KObject<any,any>[]): void {
  }

}

